h/services/group_update.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from sqlalchemy.exc import SQLAlchemyError, IntegrityError
+
+from h.exceptions import APIError, ConflictError
+
+
+class GroupUpdateService(object):
+
+    def __init__(self, session):
+        """
+        Create a new GroupUpdateService
+
+        :param session: the SQLAlchemy session object
+        """
+        self.session = session
+
+    def update(self, group, data):
+        """
+        Update a group model with the data provided.
+
+        This assumes that ``data`` has already been validated through a schema
+        or other methods. This method will intercept known SQL/Integrity errors
+        but the caller should be cautious not to expose unhandled Exceptions.
+
+        :raises ConflictError: If a provided ``authority_provided_id`` violates a
+                               DB uniqueness constraint. See :py:`h.views.api.exceptions`
+                               for a handler view for this error
+        :raises APIError:      If an unexpected SQLAlchemy error is encountered, to
+                               avoid exposing an SQL-specific Exception to users/clients
+        :raises ValueError:    (via model) if attributes fail model-level validation constraints
+        """
+
+        for key, value in data.items():
+            setattr(group, key, value)
+
+        try:
+            self.session.flush()
+
+        except IntegrityError as err:
+            if "group__groupid" in repr(err):
```

Comment:
I spent a _while_ tooling around the innards of SQLAlchemy exceptions and couldn't find a better way to introspect the errors. There were no useful properties that I could see on the exceptions. This string matching feels a bit gross...